nixos/modules/services/networking/tailscale.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.tailscale;
+in {
+  meta.maintainers = with maintainers; [ danderson mbaillie ];
+
+  options.services.tailscale = {
+    enable = mkEnableOption "Tailscale daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tailscale;
+      defaultText = "pkgs.tailscale";
+      description = "The package providing Tailscale.";
+    };
```

Comment:
Unless this is something people are likely to need, I suggest leaving it out. Overlays are always there as an escape hatch.

Reply:
Later, I would like to provide derivations for unstable builds of tailscale, built from our CI. I'm not sure how to integrate this cleanly, but at least for the NixOS machines that we run, we would like a way to easily override the package to a different derivation.

Is your suggestion that we should do this by overlaying `pkgs.tailscale` in our own NixOS configs? (Sorry if this is an obvious question, I haven't done overlays yet)

Reply:
Yes, overlays would be a fine way to do this unless either

- Somebody is likely to want multiple versions of the same package at once for some reason
- Using a non-default tailscale package is common enough that it makes sense to offer this option as a convenience.

Reply:
Thanks! Based on this, I agree we should just remove this for now, and add it back if it turns out to be a pain point.

Reply:
I've removed this now

Reply:
Relevant is https://github.com/NixOS/nixpkgs/issues/50476